code/modules/reagents/chemistry/recipes/cat2_medicines.dm

```diff
@@ -1,79 +1,372 @@
 	results = list(/datum/reagent/medicine/c2/hercuri = 5)
 	required_reagents = list(/datum/reagent/cryostylane = 3, /datum/reagent/bromine = 1, /datum/reagent/lye = 1)
-	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_HEALING | REACTION_TAG_BURN
-	required_temp = 47
+
 	is_cold_recipe = TRUE
+	required_temp = 47
 	optimal_temp = 10
 	overheat_temp = 5
-	thermic_constant = -50
+	optimal_ph_min = 6
+	optimal_ph_max = 10
+	determin_ph_range = 1
+	temp_exponent_factor = 3
+	thermic_constant = -40
+	H_ion_release = 3.7
+	rate_up_lim = 50
+	reaction_flags = REACTION_PH_VOL_CONSTANT
+	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_HEALING | REACTION_TAG_BURN
+
+/datum/chemical_reaction/medicine/hercuri/overheated(datum/reagents/holder, datum/equilibrium/equilibrium, vol_added)
+	if(off_cooldown(holder, equilibrium, 2, "hercuri_freeze"))
+		return
+	playsound(holder.my_atom, 'sound/magic/ethereal_exit.ogg', 50, 1)
+	holder.my_atom.visible_message("The reaction frosts over, releasing it's chilly contents!")
+	var/radius = max((equilibrium.step_target_vol/50), 1)
+	freeze_radius(holder, equilibrium, 200, radius, 50) //drying agent exists
+	explode_shockwave(holder, equilibrium, sound_and_text = FALSE)
 
 /*****OXY*****/
+//These react faster with optional oxygen, and have blastback effects! (the oxygen makes their fail states deadlier)
 
 /datum/chemical_reaction/medicine/convermol
 	results = list(/datum/reagent/medicine/c2/convermol = 3)
 	required_reagents = list(/datum/reagent/hydrogen = 1, /datum/reagent/fluorine = 1, /datum/reagent/fuel/oil = 1)
 	required_temp = 370
 	mix_message = "The mixture rapidly turns into a dense pink liquid."
+	optimal_temp = 420
+	overheat_temp = 720 //Ash will be created before this - so it's pretty rare that overheat is actually triggered
+	optimal_ph_min = 2
+	optimal_ph_max = 7
+	determin_ph_range = 2
+	temp_exponent_factor = 0.75
+	ph_exponent_factor = 1.25
+	thermic_constant = 15
+	H_ion_release = -1
+	rate_up_lim = 50
+	purity_min = 0.4
+	reaction_flags = REACTION_PH_VOL_CONSTANT
 	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_HEALING | REACTION_TAG_OXY
 
+/datum/chemical_reaction/medicine/convermol/reaction_step(datum/equilibrium/reaction, datum/reagents/holder, delta_t, delta_ph, step_reaction_vol)
+	. = ..()
+	var/datum/reagent/oxy = holder.has_reagent(/datum/reagent/oxygen)
+	if(!oxy)
+		reaction.delta_t = delta_t/10 //slow without oxygen
+	else
+		holder.remove_reagent(/datum/reagent/oxygen, 0.25)
+
+/datum/chemical_reaction/medicine/convermol/overheated(datum/reagents/holder, datum/equilibrium/equilibrium, impure = FALSE)
+	var/range = impure ? 4 : 3
+	if(holder.has_reagent(/datum/reagent/oxygen))
+		explode_shockwave(holder, equilibrium, range) //damage 5
+	else
+		explode_shockwave(holder, equilibrium, range, damage = 2)
+
+/datum/chemical_reaction/medicine/convermol/overly_impure(datum/reagents/holder, datum/equilibrium/equilibrium, vol_added)
+	. = ..()
+	overheated(holder, equilibrium, TRUE)
```

Comment:
You should use named arguments with non-obvious variables like this TRUE.

Reply:
Done